ambry-router/src/test/java/com.github.ambry.router/GetBlobOperationTest.java

```diff
@@ -474,6 +478,123 @@ public void testLegacyBlobGetSuccess()
+  public void testRangeRequestSimpleBlob()
+      throws Exception {
+    // Random valid ranges
+    for (int i = 0; i < 5; i++) {
+      blobSize = random.nextInt(maxChunkSize) + 1;
+      int randomOne = random.nextInt(blobSize);
+      int randomTwo = random.nextInt(blobSize);
+      testRangeRequestClosedRange(Math.min(randomOne, randomTwo), Math.max(randomOne, randomTwo), true);
+    }
+
+    blobSize = random.nextInt(maxChunkSize) + 1;
+    // Entire blob
+    testRangeRequestClosedRange(0, blobSize - 1, true);
+    // Range that extends to end of blob
+    testRangeRequestOpenRange(random.nextInt(blobSize), true);
+    // Last n bytes of the blob
+    testRangeRequestLastNBytes(random.nextInt(blobSize) + 1, true);
+    // Last blobSize + 1 bytes (should not succeed)
+    testRangeRequestLastNBytes(blobSize + 1, false);
+    // Range over the end of the blob (should not succeed)
+    testRangeRequestClosedRange(random.nextInt(blobSize), blobSize + 5, false);
+    // Ranges that start past the end of the blob (should not succeed)
+    testRangeRequestOpenRange(blobSize, false);
+    testRangeRequestClosedRange(blobSize, blobSize + 20, false);
+    // 0 byte range
+    testRangeRequestLastNBytes(0, true);
+  }
+
+  /**
+   * Test range requests on a composite blob.
+   * @throws Exception
+   */
+  @Test
+  public void testRangeRequestCompositeBlob()
+      throws Exception {
+    // Random valid ranges
+    for (int i = 0; i < 5; i++) {
+      blobSize = random.nextInt(maxChunkSize) + maxChunkSize * random.nextInt(10);
+      int randomOne = random.nextInt(blobSize);
+      int randomTwo = random.nextInt(blobSize);
+      testRangeRequestClosedRange(Math.min(randomOne, randomTwo), Math.max(randomOne, randomTwo), true);
+    }
+
+    blobSize = random.nextInt(maxChunkSize) + maxChunkSize * random.nextInt(10);
+    // Entire blob
+    testRangeRequestClosedRange(0, blobSize - 1, true);
+    // Range that extends to end of blob
+    testRangeRequestOpenRange(random.nextInt(blobSize), true);
+    // Last n bytes of the blob
+    testRangeRequestLastNBytes(random.nextInt(blobSize) + 1, true);
+    // Last blobSize + 1 bytes (should not succeed)
+    testRangeRequestLastNBytes(blobSize + 1, false);
+    // Range over the end of the blob (should not succeed)
+    testRangeRequestClosedRange(random.nextInt(blobSize), blobSize + 5, false);
+    // Ranges that start past the end of the blob (should not succeed)
+    testRangeRequestOpenRange(blobSize, false);
+    testRangeRequestClosedRange(blobSize, blobSize + 20, false);
+    // 0 byte range
+    testRangeRequestLastNBytes(0, true);
+
+    blobSize = maxChunkSize * 2 + random.nextInt(maxChunkSize);
+    // Single start chunk
+    testRangeRequestClosedRange(0, maxChunkSize - 1, true);
+    // Single intermediate chunk
+    testRangeRequestClosedRange(maxChunkSize, maxChunkSize * 2 - 1, true);
+    // Single end chunk
+    testRangeRequestClosedRange(maxChunkSize * 2, blobSize - 1, true);
+  }
```

Comment:
test cases for ranges that go across chunk boundaries? Start in the middle etc? While the random test is good, it might not guarantee testing all the cases I think